### DISCUSSION TITLE
Integration with Keycloak

### DISCUSSION THREAD

**Discussion by MrArabboy** [minio/minio#16801] [Q&A] [answered]
Hello everyone! I need help to integrate Keycloak to Minio. I configured settings like written in https://github.com/minio/minio/blob/master/docs/sts/keycloak.md . When I press "login with sso" in minio login page it redirects to keycloak and after login in keycloak it redirect minio login page again but not signed in . Please help me to fix that

**Comment by ravindk89** [comment, marked as answer]
We need a lot more information here. MinIO version, Keycloak version, and how you configured MinIO and Keycloak.

We also have an upcoming tutorial on Keycloak + MinIO - staged [here](https://github.com/minio/docs/pull/735) if you'd like to take a look.

**Reply by MrArabboy** [reply]
I set wrong Redirect URI _minio console url_  instead of **<minio console url/oauth_callback>**. So it is now working correctly. But one more question, is it possible to create minio user after login with Keycloak ?

**Reply by ravindk89** [reply]
Yes - you can do so using the MinIO CLI and `mc admin user`. GUI support for multi iDP is ongoing.
